daemon/container.go

```diff
@@ -256,6 +256,12 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 		return nil, fmt.Errorf("can't create 'AutoRemove' container with restart policy")
 	}
 
+	for _, extraHost := range hostConfig.ExtraHosts {
+		if err := opts.ValidateExtraHost(extraHost); err != nil {
```

Comment:
Oh, this validation still returns the original input string, sorry for my mistake. Updated.